src/Microsoft.OpenApi/Writers/SpecialCharacterStringExtensions.cs

```diff
@@ -134,39 +135,11 @@ internal static string GetYamlCompatibleString(this string input)
                 input = input.Replace("\"", "\\\"");
 
                 // Escape all the control characters.
-                input = input.Replace("\0", "\\0");
-                input = input.Replace("\x01", "\\x01");
-                input = input.Replace("\x02", "\\x02");
-                input = input.Replace("\x03", "\\x03");
-                input = input.Replace("\x04", "\\x04");
-                input = input.Replace("\x05", "\\x05");
-                input = input.Replace("\x06", "\\x06");
-                input = input.Replace("\a", "\\a");
-                input = input.Replace("\b", "\\b");
-                input = input.Replace("\t", "\\t");
-                input = input.Replace("\n", "\\n");
-                input = input.Replace("\v", "\\v");
-                input = input.Replace("\f", "\\f");
-                input = input.Replace("\r", "\\r");
-                input = input.Replace("\x0e", "\\x0e");
-                input = input.Replace("\x0f", "\\x0f");
-                input = input.Replace("\x10", "\\x10");
-                input = input.Replace("\x11", "\\x11");
-                input = input.Replace("\x12", "\\x12");
-                input = input.Replace("\x13", "\\x13");
-                input = input.Replace("\x14", "\\x14");
-                input = input.Replace("\x15", "\\x15");
-                input = input.Replace("\x16", "\\x16");
-                input = input.Replace("\x17", "\\x17");
-                input = input.Replace("\x18", "\\x18");
-                input = input.Replace("\x19", "\\x19");
-                input = input.Replace("\x1a", "\\x1a");
-                input = input.Replace("\x1b", "\\x1b");
-                input = input.Replace("\x1c", "\\x1c");
-                input = input.Replace("\x1d", "\\x1d");
-                input = input.Replace("\x1e", "\\x1e");
-                input = input.Replace("\x1f", "\\x1f");
-
+                foreach (var replacement in _yamlControlCharacterReplacements)
+                {
+                    input = input.Replace(replacement.Key.ToString(), replacement.Value);
```

Comment:
could we use a second dictionary or something similar to avoid the string alloc caused by the Key.ToString()?

Reply:
Thank you for your suggestion. I added an additional commit :)